src/virtual_dom/vlist.rs

```diff
@@ -104,22 +105,67 @@ impl VDiff for VList {
 
         // Process children
         let mut lefts = self.children.iter_mut();
-        let mut rights = rights.drain(..);
-        loop {
-            match (lefts.next(), rights.next()) {
-                (Some(left), Some(right)) => {
-                    previous_sibling = left.apply(parent, previous_sibling.as_ref(), Some(right));
+        let has_keys = if let Some(first) = rights.first() {
+            if let Some(_) = first.key() {
+                true
+            } else {
+                false
+            }
+        } else {
+            true
+        };
+        if has_keys {
+            let mut rights_lookup = HashMap::with_capacity(rights.len());
```

Comment:
This types the `HashMap` with an `Option<String>` as a key. I think it'd be better to use `String` as the key

Reply:
Done switched to a vec and hashmap to keep track of the nodes.